corrected use of PSModulePath casing to be consistent with Windows PowerShell

SteveL-MSFT opened the pull request:
addresses #3227 

<!--

If you are a PowerShell Team member, please make sure you choose the Reviewer(s) and Assignee for your PR.
If you are not from the PowerShell Team, you can leave the fields blank and the Maintainers will choose them for you. If you are familiar with the team, feel free to mention some Reviewers yourself.

For more information about the roles of Reviewer and Assignee, refer to CONTRIBUTING.md.

-->

SteveL-MSFT committed:
corrected use of PSModulePath casing to be consistent with Windows PowerShell
addresses #3227

lzybkr commented on src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs:
```diff
@@ -734,8 +734,8 @@ private static bool NeedToClearProcessModulePath(string currentProcessModulePath
                 // and shared module paths.
-                string hklmModulePath = GetExpandedEnvironmentVariable("PSMODULEPATH", EnvironmentVariableTarget.Machine);
-                string hkcuModulePath = GetExpandedEnvironmentVariable("PSMODULEPATH", EnvironmentVariableTarget.User);
+                string hklmModulePath = GetExpandedEnvironmentVariable("PSModulePath", EnvironmentVariableTarget.Machine);
```
I count 8 references to this same string in code - there should be a `const string` that we use instead.

SteveL-MSFT replied:
ok

iSazonov commented on test/powershell/engine/Remoting/SessionOption.Tests.ps1:
```diff
@@ -1,3 +1,4 @@
+return
```
It seems we should remove.

SteveL-MSFT replied:
that was accidentally committed as those tests were failing due to external issue, will remove

iSazonov commented on test/powershell/Language/Classes/Scripting.Classes.Modules.Tests.ps1:
```diff
@@ -76,9 +76,9 @@ Describe 'Module reloading with Class definition' -Tags "CI" {
-function Get-PassedArgsRoot { [Root]::new().passedIn }
+$passedArgs = $args
+class Root { $passedIn = $passedArgs }
+function Get-PassedArgsRoot { [Root]::new().passedIn }
```
What's changed?

SteveL-MSFT replied:
Didn't change anything here as PSModulePath isn't here.  Not sure why this is showing up in the PR as a change.

SteveL-MSFT replied:
Don't see any whitespace difference either

SteveL-MSFT committed:
addressing review feedback
make "PSModulePath" into const
fixed some test workarounds due to failures for external reasons that wasn't meant to be checked in

SteveL-MSFT committed:
addressing review feedback
make "PSModulePath" into const
fixed some test workarounds due to failures for external reasons that wasn't meant to be checked in

lzybkr commented on src/System.Management.Automation/engine/PropertyAccessor.cs:
```diff
@@ -518,11 +518,11 @@ internal override string GetModulePath(PropertyScope scope)
             if (PropertyScope.CurrentUser == scope)
             {
-                return ModuleIntrinsics.GetExpandedEnvironmentVariable("PSMODULEPATH", EnvironmentVariableTarget.User);
+                return ModuleIntrinsics.GetExpandedEnvironmentVariable("PSModulePath", EnvironmentVariableTarget.User);
```
`Constants.PSModulePathEnvVar` - right?

lzybkr replied:
Whitespace changes - add `?w=1` to see the url to see diffs ignoring whitespace.

SteveL-MSFT replied:
Did a search for `"PSModulePath"` and I think I got them all

SteveL-MSFT committed:
Merge branch 'psmodulepath' of github.com:SteveL-MSFT/PowerShell into psmodulepath

SteveL-MSFT commented:
@TravisEz13 merge?
